### PR TITLE
support sending scan summary & link

### PR DESCRIPTION
**Pull request by gnmahanth**
related to https://github.com/deepfence/enterprise-roadmap/issues/1988

### CODE REVIEW

File: deepfence_server/constants/api-messages/success.go [hunk 1]
```diff
@@ -14,4 +14,5 @@ const (
 	ErrIntegrationTypeCannotBeUpdated = "integration type cannot be updated"
 	ErrIntegrationTypeEmpty           = "integration type cannot be empty"
 	ErrNotificationTypeEmpty          = "notification type cannot be empty"
+	ErrSendSummaryNotSupported        = "notification type doesnot support sending summary links"
```
Comment (ramanan-ravi) [inline deepfence_server/constants/api-messages/success.go:17]: Typo: does not

**Review by ramanan-ravi — APPROVED** [review]

File: deepfence_server/handler/integration.go [hunk 2]
```diff
@@ -44,6 +44,18 @@ func (h *Handler) AddIntegration(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if req.SendSummary {
+		if integration.SupportsSummaryLink(req.IntegrationType) {
+			req.Config["send_summary"] = True
```
Comment (ibreakthecloud) [inline deepfence_server/handler/integration.go:49]: question: why are we getting send_summary in req and then setting "send_summary" in config, can't we directly take this in config from request?

Comment (ibreakthecloud) [reply deepfence_server/handler/integration.go:49]: or just keep it in main struct, not inside config would be easier to marshal/unmarshal

File: deepfence_server/pkg/integration/integration.go [hunk 3]
```diff
@@ -58,17 +58,32 @@ func GetIntegration(ctx context.Context, integrationType string, b []byte) (Inte
...
 }
 
+func SupportsSummaryLink(integrationType string) bool {
+	switch integrationType {
+	case constants.Slack,
+		constants.Teams,
+		constants.Email:
+		return true
+	}
+	return false
```
Comment (ibreakthecloud) [inline deepfence_server/pkg/integration/integration.go:78]: nit: I think we should maintain a array of supported in constants. feel free to ignore this comment

File: deepfence_server/handler/integration.go (hunk 2 above)
Comment (gnmahanth) [reply deepfence_server/handler/integration.go:49]: this config doesn't apply for all integrations and keeping in main struct would required add a new column in the postgresql table

File: deepfence_server/pkg/integration/integration.go (hunk 3 above)
Comment (gnmahanth) [reply deepfence_server/pkg/integration/integration.go:78]: will keep this for now

**Review by ibreakthecloud — APPROVED** [review]

**Pull request merged**